### PR TITLE
Updated travis and added go mod.

### LINKED ISSUE #77: Add support for Go Modules
Since 1.11 Go Modules is probably the preferred way to handle versioning.
We should add support for it!

### PR DESCRIPTION
**Pull request by elliotcourant**
This is an attempt at adding go mod to this library.

@ahmetb I'm not sure if you had any plans on adding this, but I've started the process to migrate some of my projects from dep to go mod and thought I'd add this too.

Should resolve #77

### CODE REVIEW

**Comment by elliotcourant** [comment]
It might also make sense if you do merge this to do a `v3.0.1` release to make tagging this library as a dependency easier. That way it will include my other change from last night too. Right now I believe go mod will default to the last release regardless of how far ahead the master branch is.

**Comment by cleitonmarx** [comment]
@ahmetb I think we can merge this one. Thoughts?

**Comment by ahmetb** [comment]
We need to add these to .travis.yaml for Go modules to actually work:

```yaml
install: true # <- so that it's a noop
env:
- GO111MODULE: on # <- because when GOPATH is set (travis does), modules are not used.
```

File: go.mod [hunk 1]
```diff
@@ -0,0 +1,3 @@
+module github.com/ahmetb/go-linq
```
Comment (ahmetb) [inline go.mod]: you sure this shouldn't have the prefix .../go-linq<b>/v3</b>?

Comment (elliotcourant) [reply go.mod]: It should yes

**Comment by elliotcourant** [comment]
I can make those changes tonight 👍

**Comment by elliotcourant** [comment]
Looks like Go 1.11 didn't handle the GO111MODULE env variable. But dependencies still get resolved automatically the first time the program is built or tested so the pipeline passes without it. 

I also added `tip` to the go version to make sure new versions work as well as they are released.

**Comment by ahmetb** [comment]
Go 1.11 handles the GO111MODULE env var for sure.

**Comment by elliotcourant** [comment]
I just looked at the travis build, I think I had done the env variable wrong. Trying something quick.

**Comment by elliotcourant** [comment]
It should now be good to go 👍

**Comment by ahmetb** [comment]
Thanks. I'll merge this but can you also help us with these:

1. The import/go-get instructions in the readme still work for projects using GOPATH and not go modules (go 1.10 or older).
2. The import/go-get instructions in the readme still work for projects using go modules.
3. Should we stick with gopkg.in in the import path for Case 2 above? I suspect not, because the module name has changed and we should update the readme.

**Comment by elliotcourant** [comment]
I can submit another MR with some readme updates tonight or tomorrow if you'd like.

Personally I prefer the github.com path (or the path to the actual repo). Just because it makes the package incredible easy to find when it's being used by someone else, and I really like how all of the packages get organized by their actual repository paths in the GOPATH. But go mod does move away from that model and it might not make sense anymore?

That's just my personal preference though.

**Comment by ahmetb** [comment]
Yeah for projects using go modules, whatever appears in go.mod is the import path.
We just need to make sure it’s not breaking old projects not using go.mod though.

**Comment by ahmetb** [comment]
Looks like I need to make a git tag with this go.mod for /v3 import path to work.

**Comment by ahmetb** [comment]
I already updated the README and made a v3.1.0 and confirmed it now works with go modules.

**Pull request merged**